v4/export/config.go

```diff
@@ -172,8 +172,8 @@ func DefaultConfig() *Config {
 // String returns dumpling's config in json format
 func (conf *Config) String() string {
 	cfg, err := json.Marshal(conf)
-	if err != nil {
-		log.Error("fail to marshal config to json", zap.Error(err))
+	if err != nil && conf.Logger != nil {
```

Comment:
didn't find where `conf.Logger` is set

Reply:
`conf.Logger` is set in DM, outside dumpling repository.